src/aws/osml/model_runner/api/image_request.py

```diff
@@ -21,148 +22,242 @@
-                             attributes
-        :param kwargs: Any = keyword arguments provided on the constructor to set specific attributes
+        :param initial_data: Dictionaries containing attributes and their values to set on the instance.
+        :param kwargs: Additional keyword arguments to set specific attributes on the instance.
         """
-        default_post_processing = [
+        self._initialize_attributes()
+        self._set_attributes_from_data(*initial_data, **kwargs)
+
+    def _initialize_attributes(self):
+        """
+        Initializes default attributes for an ImageRequest instance.
+        """
+        self.tile_size = DEFAULT_TILE_SIZE
+        self.tile_overlap = DEFAULT_TILE_OVERLAP
+        self.tile_format = ImageFormats.NITF
+        self.tile_compression = ImageCompression.NONE
+        self.post_processing = [
             MRPostProcessing(step=MRPostprocessingStep.FEATURE_DISTILLATION, algorithm=FeatureDistillationNMS())
         ]
+        self.outputs = []
 
-        self.job_id: str = ""
-        self.image_id: str = ""
-        self.image_url: str = ""
-        self.image_read_role: str = ""
-        self.outputs: List[dict] = []
-        self.model_name: str = ""
-        self.model_invoke_mode: ModelInvokeMode = ModelInvokeMode.NONE
-        self.tile_size: ImageDimensions = (1024, 1024)
-        self.tile_overlap: ImageDimensions = (50, 50)
-        self.tile_format: ImageFormats = ImageFormats.NITF
-        self.tile_compression: ImageCompression = ImageCompression.NONE
-        self.model_invocation_role: str = ""
-        self.feature_properties: List[dict] = []
-        self.roi: Optional[BaseGeometry] = None
-        self.post_processing: List[MRPostProcessing] = default_post_processing
-
-        for dictionary in initial_data:
-            for key in dictionary:
-                setattr(self, key, dictionary[key])
-        for key in kwargs:
-            setattr(self, key, kwargs[key])
+    def _set_attributes_from_data(self, *initial_data: Dict[str, Any], **kwargs: Any):
+        """
+        Sets attributes on the instance from initial data dictionaries and keyword arguments.
 
-    @staticmethod
-    def from_external_message(image_request: Dict[str, Any]):
+        :param initial_data: Dictionaries of attribute values.
+        :param kwargs: Keyword arguments to set attributes.
         """
-        This method is used to construct an ImageRequest given a dictionary reconstructed from the
-        JSON representation of a request that appears on the Image Job Queue. The structure of
-        that message is generally governed by AWS API best practices and may evolve over time as
-        the public APIs for this service mature.
+        for data in initial_data:
+            self._set_attributes_from_dict(data)
+        self._set_attributes_from_dict(kwargs)
+
+        # Ensure `post_processing` defaults if explicitly set to None
+        if self.post_processing is None:
```

Comment:
I'm not sure this is the intended behavior.  If 'post_processing' is not in the dictionary/request it should default.  If it is present in the request and set to None or [] that means the user is requesting MR to skip the post-processing steps.

Reply:
It was ensuring we are wetting one if we set the passed parameters to None versus not setting it at all.